update time to <1.15

geekosaur opened the pull request:
### Description

Update bounds of `time` to allow the new 1.14 release.

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I've considered how to best test these changes (property, unit,
        manually, ...) and concluded: build with `--constraint='time == 1.14'` passed locally

  - [ ] I updated the `CHANGES.md` file (n/a)

geekosaur commented:
Actually, since `time` is a bootlib CI is still using 1.12.2. Have to see how to test it locally.

liskin commented:
I believe @slotThe did the testing a couple times in the past when I pinged him on IRC after the Packdeps CI failed.
Perhaps we should add a note about this to MAINTAINERS.md so we don't need to remember (and also to decrease the bus factor).

slotThe commented:
I saw the 

> build with --constraint='time == 1.14' passed locally

edit and figured that the local tests ran through.

> We could even make the Packdeps job point us to the relevant piece of documentation to minimise the brain function required to deal with this :-D

We could also look into using https://github.com/nomeata/cabal-plan-bounds to generate version constraints automatically, or https://github.com/nomeata/cabal-force-upper-bound to force CI to use the upper bound of a package

geekosaur commented:
> edit and figured that the local tests ran through.

They did, and I updated to show it. Should I have deleted that comment afterward? (or edited it, I guess)
